### ISSUE TITLE
Umbraco 8.18.3 reading images from azure blob storage, writing images to local /media && 404 when not using query string ?width=xxx

### ISSUE THREAD

**Issue by elliottjb7** [JimBobSquarePants/UmbracoFileSystemProviders.Azure#212]
Umbraco 8.18.3
UmbracoFileSystemProviders.Azure 3.02

Hi, I recently upgraded from V7 to V8. Having previously used this Nuget package to host images on AzureBlobStorage, I know our setup works. However I'm getting some strange issue with the new installation. Hopefully this is my ineptitude and not a bug, but I can't get any further forward with this.

// Issue 1 - Locally
When I log in to backoffice, I can see all of my files from blob storage. Success!!
When I upload a new image via back office I expect it to be pushed up to blob storage, but I can see from version control that it's being added to the local file structure. Fail!!

// Issue 2 - locally frontend
Weirdly none of my images are displaying on the frontend. Upon investigation I found that it's because the image url is missing ?width=xxx. In backoffice all of the images are appended with the query string "localhost:12345/media/image-url.png?width=xxx - however, these images return a 404 when I remove the ?width=xxx meaning that as per original setup none of my frontend images work.

Can you help? Is this a bug?

Thanks.

**Comment by arnaud-bertrand** [comment]
I've have the same issue and it's been driving me crazy for weeks. 

For me everything works as expected locally and on my development environment on Umbraco Cloud. But for some reasons it doesn't on my Live Environment: I always get a 404 on images unless there is a query string.

It sometimes gets fixed by reinstalling the package via NuGet: it then works during a few days and then it randomly doesn't work anymore again, the 404 errors come back.

**Comment by SimonAntony** [comment]
@arnaud-bertrand @elliottjb7  Did you guys ever get around these errors? I've got the same on a v8 site (upgraded from v7), old images show ok, cannot upload new to blob (just go local instead) and all config is as per instructions. Going to grab the source and see if I can trace it through that way

**Comment by arnaud-bertrand** [comment]
Never managed to solve it but strangely enough if I reinstall the package ImageProcessor.Web.Plugins.AzureBlobCache on local and then deploy that to my development and live environment, it fixes it for a little while (sometimes only after several reinstalls and deploys). And then the error appears again after a few weeks. Beyond me why.

**Comment by SimonAntony** [comment]
> Never managed to solve it but strangely enough if I reinstall the package ImageProcessor.Web.Plugins.AzureBlobCache on local and then deploy that to my development and live environment, it fixes it for a little while (sometimes only after several reinstalls and deploys). And then the error appears again after a few weeks. Beyond me why.

Strange. I've just grabbed the source and seeing if I can attach and trace through - will update here if I find anything.

**Comment by SimonAntony** [comment]
@arnaud-bertrand Found the issue. Missing DLL in the nuget package so it does not even load! Basically the file Our.Umbraco.FileSystemProviders.Azure.Media.dll is missing which has the composer within and loads the initial config. This is missing so is not initialised. I downloaded the source, built it, grabbed a copy of the built dll, added to bin folder and it all started working fine. /cc @Jeavon not sure how I can do a PR to fix this as it's a part of the build process for the nuget package?

**Comment by Jeavon** [comment]
Did you install? https://www.nuget.org/packages/UmbracoFileSystemProviders.Azure.Media/

**Comment by SimonAntony** [comment]
> Did you install? https://www.nuget.org/packages/UmbracoFileSystemProviders.Azure.Media/

Nope! Didnt realise it was required as assumed that installing the main azure package would bring in everything else needed. Will try again, thanks.
